databases/mariadb/files/patch-powerpc.diff

```diff
@@ -0,0 +1,11 @@
+--- a/storage/xtradb/include/os0sync.h	2020-05-06 18:54:23.000000000 +0800
++++ b/storage/xtradb/include/os0sync.h	2024-05-23 08:10:13.000000000 +0800
+@@ -345,7 +345,7 @@
+ /**********************************************************//**
+ Returns the old value of *ptr, atomically sets *ptr to new_val */
+ 
+-#if defined(__powerpc__) || defined(__aarch64__)
++#if defined(__powerpc__) || defined(__POWERPC__) || defined(__aarch64__)
```

Comment:
> Did anyone build it for ppc, like ever? )

Probably, but not necessarily on a PowerPC Mac.

> It is trivially fixable, but without this the code cannot possibly compile, I believe.

The code you changed already checked if `__powerpc__` was defined; presumably that worked for whoever added it. On your PowerPC system, that wasn't defined but `__POWERPC__` was defined. According to [this page](https://sourceforge.net/p/predef/wiki/Architectures/) there are ten different macros that might be defined on various PowerPC systems but not all of the marcros are defined on all such systems. You may find that there are other places in the code that check for `__powerpc__` and you probably want to change those as well, otherwise it's possible that although it compiles on a PowerPC Mac it doesn't work correctly.

Note that this port is for the 5.5 branch of mariadb which reached [end of life](https://mariadb.com/kb/en/mariadb-5568-release-notes/) four years ago. You may find similar fixes needed in the newer versions of mariadb available in the ports mariadb-10.0 through mariadb-10.11.

Reply:
@ryandesign Thank you, I will check for those. `__powerpc__` is a linuxism and will not work on macOS.

Reply:
Rebuilding now with fixed macros, once successful, gonna mark the PR as ready.